aspnet/mvc/models/formatting.rst

```diff
@@ -1,8 +1,217 @@
+
+Browsers and Content Negotiation
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Unlike typical API clients, web browsers tend to supply ``Accept`` headers that include a wide array of formats, including wildcards. By default, when the framework detects that the request is coming from a browser, it will ignore the ``Accept`` header and instead return the content in the application's configured default format (JSON unless otherwise configured). This provides a more consistent experience when using different browsers to consume APIs.
+
+If you would prefer your application honor browser accept headers, you can configure this as part of MVC's configuration by setting ``RespectBrowserAcceptHeader`` to ``true`` in the ``ConfigureServices`` method in *Startup.cs*.
+
+.. literalinclude:: formatting/sample/src/ResponseFormattingSample/Startup.cs
+  :language: c#
+  :lines: 30-34,36,39
+  :emphasize-lines: 5
+  :dedent: 8
+
+Configuring Formatters
+----------------------
+
+If your application needs to support additional formats beyond the default of JSON, you can add these as additional dependencies in *project.json* and configure MVC to support them. There are separate formatters for input and output. Input formatters are used by :doc:`model-binding`; output formatters are used to format API responses. You can also configure :doc:`custom-formatters`.
+
+Configuring the JSON Formatter
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+A common request when formatting responses using JSON is to use JavaScript-style "camel" casing (the first character of each property is lowercase, then uppercase is used to begin each additional word making up the name). Support for camel casing can be added by using the ``AddJsonOptions`` extension method when configuring MVC:
+
+.. code-block:: c#
+
+  services.AddMvc()
+    .AddJsonOptions(o =>
+      o.SerializerSettings.ContractResolver = 
+        new CamelCasePropertyNamesContractResolver());
+
+Replace the default ``SerializerSettings.ContractResolver`` with a new instance of ``CamelCasePropertyNamesContractResolver``.
+
+Adding XML Format Support
+^^^^^^^^^^^^^^^^^^^^^^^^^
+
+To add support for XML formatting, add the "Microsoft.AspNetCore.Mvc.Formatters.Xml" package to your *project.json*'s list of dependencies.
+
+.. literalinclude:: formatting/sample/src/ResponseFormattingSample/project.json
+  :language: javascript
+  :lines: 4-26
+  :emphasize-lines: 8
+  :dedent: 2
+
+Next, add the ``XmlSerializerOutputFormatter`` to MVC's list of OutputFormatters in *Startup.cs*:
+
+.. literalinclude:: formatting/sample/src/ResponseFormattingSample/Startup.cs
+  :language: c#
+  :lines: 30-39
+  :emphasize-lines: 6
+  :dedent: 8
+
+Alternately, you can just call the ``AddXmlSerializerFormatters`` extension method:
```

Comment:
Alternately, call the ``AddXmlSerializerFormatters`` extension method:

This is probably the preferred approach - if so, show it first.